### ISSUE TITLE
new: tabs

### ISSUE THREAD

**Issue by jackreeves** [telus/tds-community#35]
## Problem Statement
We currently have varying styles of tabs, in-page navigation and filtering throughout TELUS.com. I’d like to build on the existing tabs design, which is only to be used as tabs (not filtering, not in-page.) This does not follow UX best practice, so I'd like to correct that. 

## Recommendation
Build on the existing tab structure for a better desktop and mobile experience. The active state will have the purple underline like the other nav conventions.

## Design intent
* Create a TDS approved tabs design for mobile and desktop
* Create a UX best practice tabs design

## Design + Dev
* Willing to design solution: Yes (Elena Verling)
* Willing to develop solution: Yes (Andrew Lam)

**Comment by varunj90** [comment]
closing as its duplicate, please refer to #200

**Comment by varunj90** [comment]
Issue reopened as Halo/BRC team will be building this as a community component. thank you @elenaverling and @Andrew-K-Lam for taking this on, looking forward to working together

**Comment by varunj90** [comment]
There is a design jam scheduled for July 30: Design Jam: Tabs, Filters, Other Navigation

Prerequisite: Go over https://docs.google.com/presentation/d/1rxgXYgXVVA7VJc-hvctoHs3hIwj0_cchDa96Iuv0fYE/edit#slide=id.p4 

Outcomes 
• Defining the differences amongst these components 
• Looking into Elena’s ask (so we unblock her) 
• Action items to update individual community tickets

**Comment by twirlingsky** [comment]
Here are the XL & XS design Home Solutions implemented sub-navigation. 
[XL] https://telus.invisionapp.com/d/main#/console/17553495/370407325/preview
[XS] https://telus.invisionapp.com/d/main#/console/17602419/370408129/preview

And the sketch file can be downloaded from the InVision project assets section.

**Comment by elenaverling** [comment]
I've updated the tab designs for mobile and desktop. They can be found here: https://telus.invisionapp.com/share/2RTC6CAZVKE

**Comment by DanGenuardi** [comment]
Here is what we are doing within the Usage page on My TELUS. This is a selection tab to toggle through different plans within a single BAN. This is a view for 6 or less selections.

![xl - usage default](https://user-images.githubusercontent.com/39997967/63185823-f7cad200-c028-11e9-826e-868ce07f27be.png)
![xs - usage default_Multi-BAN Copy](https://user-images.githubusercontent.com/39997967/63185834-fbf6ef80-c028-11e9-8bda-dc058df0edf7.png)

**Comment by varunj90** [comment]
@elenaverling and @DanGenuardi to jam on the mobile version for tabs. sweet!

**Comment by elenaverling** [comment]
@donnavitan @varunj90 @Christina-Lo I've added the artwork and InVision links above for mobile, tablet and desktop, based on the notes from Sept 5th: https://docs.google.com/document/d/1WOXhc53MlCcH9mV5HEhvcT8t1rNa7eQr94EcoG1rGk0/edit#

**Comment by lynnepeacock** [comment]
@elenaverling 
Could you let me know what the expected interaction should be for labeling the tab content when it's in the dropdown format?

For example: We are showing a label outside of the drop-down container in Usage, below

<img width="309" alt="Screen Shot 2019-10-03 at 12 51 38 PM" src="https://user-images.githubusercontent.com/41124620/66147293-9365d100-e5dc-11e9-8a92-e42a752a1051.png">

1) Where we say "Select a plan", would this be the default label for the drop-down?
2) When a selection has been made and the drop-down is closed, what does this label show? 
Is it now the selected value? Will a customer remember what the drop-down contains/represents in this state. I'd like to see these states in action if that's possible?

**Comment by DanGenuardi** [comment]
So from our discussion for the tab selection. It will always default to the first selection on the tab list. Then the user will manually need to select the other tabs to view within the page. The selected state will always appear on the list for mobile dropdown view.

**Comment by theetrain** [comment]
DPA discussion, let's consider:

* Not providing a fixed width for each individual tab (provide a max-width, and have it fit its contents)
* Explore truncation; when it should occur, and how
* Technical implementation: should it accept `<br />`? 
* Should designers be allowed to include icons in tabs?
* How many tabs is too many? As much as the viewport can handle?

cc @DanGenuardi for user testing insights

**Comment by elenaverling** [comment]
I've written out a very detailed recommendation list for mobile desktop and tablet after reviewing with Donna today, please read the above description text.

**Comment by varunj90** [comment]
ticket update
- design is done, Elena and Lynne have added context to the interaction storywriting 
- Halo team has deprioritized the work w.r.t this component
- a dev and an interaction designer need to take this forward

**Comment by twirlingsky** [comment]
Ivan Correces is picking up the IxD needed on this ticket. I will figure out how to add him.

**Comment by varunj90** [comment]
@gkychow has done some visual design updates, and Ivan is picking it from an iXD perspective.

**Comment by twirlingsky** [comment]
InVision project: https://telus.invisionapp.com/d/main#/projects/prototypes/18077659HA 
HS Catalogue Ticket: https://telusdigital.atlassian.net/browse/CATA-1892

**Comment by twirlingsky** [comment]
https://telus.invisionapp.com/share/2RTC6CAZVKE#/392910795_XL_Updated_-_1200px

**Comment by twirlingsky** [comment]
![screencapture-telus-invisionapp-d-main-2019-11-18-14_38_03](https://user-images.githubusercontent.com/22402633/69100010-11592c80-0a11-11ea-8003-e66c7c93f6a1.png)

**Comment by donnavitan** [comment]
1. Is the main description updated with the latest story writing acceptance criteria? I noticed that the updated designs fixed width which is counter to the best practices and initial recommendation. What's the rationale for providing fixed width? Won't this further limit the flexibility of having  more or less menu items? As well make the user travel further along the x-axis more than necessary, especially if you're a zoomed-in user.
2. From what I understand, the design is meant to expand 100% of the viewport (horizontal line with Athens Grey background) with the tabs horizontally centered? Or does it left-align to the content width?
3. Is there a recommended limit of how many items should be in the menu while using the "chevron" to navigate? If MyTELUS is/will use this component for the purpose of navigating between accounts then you can test the usability of having to navigate through that? Is that the use case this component should allow?
4. Is the tabs component meant to expand all the way to the bottom of the page (footer not included)? It will never be used to switch between content in a small area of the page and have other content below it that is not part of the active tab? It does not act like the ExpandCollapse component in a contained way?
5. What is the difference between this component and an in-page navigation other than being used further down the page? As per [Pik TV example](https://telus.invisionapp.com/d/main#/console/18077659/393995674/preview). This usage seems counter to best practices and recommendation.

**Comment by twirlingsky** [comment]
For #2. Gary designed the tabs to fill the content width with an inset spacing of level 4. (So regular 16px margin, level 4 space, then the tabs.

The Pik TV catalogue example isn't in-page navigation - it calls the offer case and filters the catalogue depending on the desired catalogue state. 

Tabs don't work like expand/collapse...

Ivan wrote up all the guidelines in the InVision doc. In there is recommended best practice for #of tabs per breakpoint and fixed widths: https://telus.invisionapp.com/share/2RTC6CAZVKE#/screens/393996282_TDS_Tabs_Usage-Guidelines

**Comment by lynnepeacock** [comment]
Hi @twirlingsky (can't seem to tag Ivan),

Thanks so much for sharing! Exciting to see the progress of the tabs. I have made a couple of comments in invision about some visual elements that I noticed. 

I have an additional question that I thought should be posted here, as there may be other people who were involved in the decision.

I noticed that the sub-copy / phone number underneath the name does not pass accessibility [contrast checks for AA](https://webaim.org/resources/contrastchecker/?fcolor=71757B&bcolor=F7F7F8). Was this a conscious decision?

I would also like to know the rules around the grey background on the tabs and when it should or shouldn't extend to the edges of the page.

Thanks :D

**Comment by twirlingsky** [comment]
I believe @gkychow set the tabs to be Shuttle grey on a gainsboro grey background (not Raven's grey text). I'll double-check the file when he send his updates. 
![Screen Shot 2019-11-19 at 09 40 52](https://user-images.githubusercontent.com/22402633/69171477-e3c3c000-0ab0-11ea-9123-823ca8eab0ca.png)

The gainsboro grey background only extends to where the tabs are - no further extension.

The focus state is indicated in the InVision project. Gary has been working on design patterns for these so that we can move away from the blue.

**Comment by Christina-Lo** [comment]
Made an update to the ticket with the latest story (in google doc) and design files
